src/sentry/db/mixin.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import absolute_import
+
+import six
+
+
+from uuid import uuid4
+
+from sentry.models import OrganizationOption
+
+
+def delete_pending_deletion_option(instance, **kwargs):
+    if not issubclass(instance.__class__, PendingDeletionMixin):
+        return
+    instance.delete_pending_deletion_option()
+
+
+class PendingDeletionMixin(object):
+    _rename_fields_on_pending_delete = []
+
+    def build_pending_deletion_key(self):
+        return 'pending-delete:%s:%s' % (self.__class__, self.id)
+
+    def rename_on_pending_deletion(self, fields=None):
+        fields = fields or self._rename_fields_on_pending_delete
+        original_data = {}
+
+        for field in fields:
+            original_data[field] = getattr(self, field)
+            setattr(self, field, uuid4().hex)
+
+        self.save()
+        original_data['id'] = self.id
+        original_data['model'] = self.__class__
```

Comment:
This is the actual class object that'll end up being serialized. We should only store a string to represent the model. I'm not sure wehre off the top of my head, but Django has a way to transform something in the form of `{app}.{Model}` into a real model. So something like, `sentry.Organization` into the `Organization` class. I'd recommend usinog that instead of the class itself, otherwise, you're going to get into issues in the future.